src/components/CustomCall.tsx

```diff
@@ -38,47 +42,50 @@ const CustomCall = () => {
   const { createType } = useApiCalls();
 
   const { isButtonDisabled, sendLaneMessage } = useSendMessage({
-    input: customCallInput,
+    input: customCallDebouncedInput,
     isValidCall: Boolean(decoded),
     type: TransactionTypes.CUSTOM,
-    weightInput
+    weightInput: weightDebouncedInput
   });
-  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    decodePayload(event.target.value);
-    setCustomCallInput(event.target.value);
-  };
-
-  const onWeightChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    setWeightInput(event.target.value);
-  };
-
-  function decodePayload(input: string) {
-    try {
-      setError(null);
+  const onChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      const input = event.target.value;
+      try {
+        setError(null);
+        setCustomCallInput(input);
+        //@ts-ignore
+        const call = createType(targetChain, 'Call', input);
+        setDecoded(JSON.stringify(call, null, 4));
+      } catch (e) {
+        setError('Wrong call provided');
```

Comment:
Might be good to at least log the original error to the console.
```suggestion
      } catch (e) {
        logger.error('Wrong call', e);
        setError('Wrong call provided');
```